src/NServiceBus.Core/Sagas/ActiveSagaInstance.cs

```diff
@@ -110,8 +124,13 @@ void AttachEntity(IContainSagaData sagaEntity)
         internal void MarkAsNotFound()
         {
             NotFound = true;
+            MarkAsModified();
         }
 
+        internal void MarkAsModified()
```

Comment:
it's kinda weird that `MarkAsModified` is internal while `Modified` has a public setter?